CoreUtilities/src/au/gov/asd/tac/constellation/utilities/query/QueryEvaluator.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2010-2019 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.utilities.query;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Stack;
+
+/**
+ * Evaluates a query expression made up of a chain of operations of the form (a
+ * op1 b op2 c ...).
+ * <p>
+ * Based on
+ * https://www.geeksforgeeks.org/stack-set-4-evaluation-postfix-expression/
```

Comment:
👍

Reply:
Is that the correct way to attribute code or should it go into Attributions?

Reply:
That is a good point. I think it would be nice to mention something in `ATTRIBUTIONS.md` too. This helps keep it all consolidated.

Reply:
Would this be suitable?
our file name, lisence and link to the licence. 
![image](https://user-images.githubusercontent.com/58677759/79300727-b9bede80-7f26-11ea-9e7d-2d82677974c8.png)

I could only find the license info on this link: https://github.com/ameyyadav09/geeksforgeeks#license

I can push this up as soon as you give the go ahead. Then I'm good for it to be merged in. @arcturus2

Reply:
that's great @aldebaran30701, please push it when you can.

Reply:
Done